NDVItimeseries.R

```diff
@@ -0,0 +1,527 @@
+
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("_NDVI", "", colnames(NDVIadmin2)[col])
+}
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("X", "", colnames(NDVIadmin2)[col])
+}
+
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("_NDVI", "", colnames(NDVIadmin3)[col])
+}
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("X", "", colnames(NDVIadmin3)[col])
+}
+head(NDVIadmin3)
+
+
+# Convert data from Wide to Long ---- 
+ndviData2 <- 
+  NDVIadmin2 %>%
+  select(admin1Name, admin1Pcod, 
+         admin2Name, admin2Pcod, "19810701": "20191231") 
+
+ndviDataLong2 <- 
+  gather(ndviData2, 
+         key = Date, 
+         value = ndvi, 
+         "19810701":"20191231") %>% 
+  mutate(Date = ymd(Date)) #convert from char to dates
+# "19810701":"20191231") 
+
+ndviDataLong2$ndvi <- as.numeric(ndviDataLong2$ndvi) 
+ndviDataLong2 <- ndviDataLong2 %>% 
+  mutate(newNDVI = 0.0001 *ndvi)
+
+ndviData3 <-
+  NDVIadmin3 %>%
+  select(adm_03, rowcacode3,  
+         "19810701":"20191231")
+
+ndviDataLong3 <-
+  gather(ndviData3,
+         Date,
+         ndvi,
+         "19810701":"20191231") %>% 
+  mutate(Date = ymd(Date)) # Create new col with dates instead of chars
+
+ndviDataLong3$ndvi <- as.numeric(ndviDataLong3$ndvi) 
+ndviDataLong3 <- ndviDataLong3 %>% 
+  mutate(newNDVI = 0.0001 *ndvi)
+
+data_admin2 <- 
+  ndviDataLong2 %>%
+  mutate(year = year(Date), 
+         month = month(Date), 
+         day = day(Date)) %>%
+  ungroup()
+
+data_admin3 <- 
+  ndviDataLong3 %>%
+  mutate(year = year(Date), 
+         month = month(Date), 
+         day = day(Date)) %>%
+  ungroup()
+
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Generate Total NDVI Graphics By Admin 0 (country) and Admin 1 (building off admin 2) ---- 
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Admin 0 (one line for the entire region)
+peakndvi <- 
+  yearData_admin2 %>%
+  group_by(year) %>%
+  mutate(meanAnnualndvi = mean(peak_ndvi)) %>%
+  ungroup() %>%
+  distinct(year, meanAnnualndvi)
+
+peakndvi %>% 
+  ggplot(aes(x=year, y=meanAnnualndvi)) +
+  geom_line() + 
+  labs(title="Peak NDVI by Year", x="Year", y ="Peak NDVI") +
+  theme_classic() 
+
+# Calculate total admin 2 and 3 ndvi for maps (if every obs is on admin 2 basis, then show)
+yearData_admin2 <- 
+  data_admin2 %>% 
+  group_by(admin2Name, admin2Pcod, year) %>%
+  mutate(peak_ndvi = max(newNDVI, na.rm = TRUE),
+         mean_ndvi = mean(peak_ndvi, na.rm = TRUE),
+         standardDeviation_ndvi = sd(peak_ndvi, na.rm = TRUE)) %>%
+  distinct(admin1Name, admin1Pcod,admin2Name, admin2Pcod, year, 
+           peak_ndvi, mean_ndvi, standardDeviation_ndvi) %>%
+  ungroup()
+
+yearData_admin3 <- 
+  data_admin3 %>% 
+  group_by(adm_03, rowcacode3, year) %>%
+  mutate(peak_ndvi = max(newNDVI, na.rm = TRUE),
+         mean_ndvi = mean(peak_ndvi, na.rm = TRUE),
```

Comment:
see comments above from admin2 -- not clear this is generating what you think it is. good place to do a check/test to see if values are in a reasonable range (and if values are distinct from one another)